Makefile

```diff
@@ -114,11 +116,15 @@ cformat:
 fix_cformat:
 	bash tools/run_clang_format.sh apply
 
-# The library file is managed by a different build tool - pretend it's always dirty.
-.PHONY: $(C_LIB_CARGO_PATH)
-$(C_LIB_CARGO_PATH):
+$(C_LIB_DEBUG_CARGO_PATH):
+	cargo rustc --crate-type cdylib -p qiskit-cext
+
+$(C_LIB_RELEASE_CARGO_PATH):
 	cargo rustc --release --crate-type cdylib -p qiskit-cext
 
+$(C_QISKIT_H):
+	cbindgen --crate qiskit-cext --output $(C_DIR_INCLUDE)/qiskit.h --lang C
```

Comment:
We probably shouldn't have duplicated targets in the Makefile, running this locally also gives some warnings. I'm not quite sure how to cleanly resolve this, given that both in the debug and release mode the header is called `target/qiskit.h`. Maybe the simple solution would be to just add the full build into the `ctest` target and rely on rust not redundantly recompiling?

Reply:
I think the simplest solution is actually to split out the cheader job to just call cbindgen manually. ctest depends on compilation already with `C_LIB_DEBUG_CARGO_PATH` so we get the header created from that, so depending on the target that relies on `cp` works fine. I pushed up the change to do this: https://github.com/Qiskit/qiskit/pull/14046/commits/a8c465f64d80ca701dac7458c7446cc515d8c45d